### PR TITLE
Add client cert in insecure mode

### LINKED ISSUE #10803: Database TLS "insecure" mode does not generate client certificate
### Description

**What happened**:

With the following database configuration:

```yaml
db_service:
  enabled: "yes"
  databases:
  - name: "postgres"
    description: "🐘 PostgreSQL 13.0: Local"
    protocol: "postgres"
    uri: "localhost:5432"
    tls:
      mode: insecure
```

Connection to Postgres fails with:

```
Original Error: *pgconn.connectError failed to connect to `host=localhost user=postgres database=postgres`: server error (FATAL: connection requires a valid client certificate (SQLSTATE 28000))
```

In the Postgres logs:

```
2022-03-03 19:29:31.600 GMT [50] FATAL:  connection requires a valid client certificate
```

It looks like TLS "insecure" mode just sets InsecureSkipVerify to true and does not actually generate client certificate for the connection:

https://github.com/gravitational/teleport/blob/v9.0.0-beta.1/lib/srv/db/common/auth.go#L384-L393

The setting should only apply to validating the _server_ certificate.

**What you expected to happen**:

Connection to self-hosted database which presents an invalid certificate should still succeed with "insecure" mode.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Configure a self-hosted Postgres with an invalid certificate.
2. Configure database access with TLS insecure mode.
3. Try to tsh db connect to it.

### PR DESCRIPTION
**Pull request by jakule**
Currently Database Access TLS `insecure` mode doesn't include client certificate when connecting to a database. This change adds client certificate to requests in `insecure` mode.
 
Closes https://github.com/gravitational/teleport/issues/10803

### CODE REVIEW

**Review by codingllama — APPROVED** [review]
Please add a brief description to the PR, but otherwise looks alright.

**Review by hatched — APPROVED** [review]

**Review by russjones — APPROVED** [review]

File: lib/srv/db/common/auth.go [hunk 1]
```diff
@@ -381,15 +381,18 @@ func (a *dbAuth) getTLSConfigVerifyFull(ctx context.Context, sessionCtx *Session
 
 // getTLSConfigInsecure generates tls.Config when TLS mode is equal to 'insecure'.
 // Generated configuration will accept any certificate provided by database.
-func getTLSConfigInsecure() *tls.Config {
-	tlsConfig := &tls.Config{
-		RootCAs: x509.NewCertPool(),
+func (a *dbAuth) getTLSConfigInsecure(ctx context.Context, sessionCtx *Session) (*tls.Config, error) {
+	tlsConfig, err := a.getTLSConfigVerifyFull(ctx, sessionCtx)
```
Comment (r0mant) [inline lib/srv/db/common/auth.go:385]: Do we really need to build a full TLS verification config first if we're going to disable verification anyway? Wouldn't just adding "appendClientCert()" work?

Comment (jakule) [reply lib/srv/db/common/auth.go:385]: We probably could, but then we have different behaviour for cloud dbs. Currently we don't include the client certificate in other modes: https://github.com/gravitational/teleport/blob/bfaf3f8228cdfa88febbd0fba094488cbda601f1/lib/srv/db/common/auth.go#L372-L374

Comment (r0mant) [reply lib/srv/db/common/auth.go:385]: Actually, you're right, but it seems like that below we're adding client cert in any case? Even when connecting to a cloud database with "insecure" flag set?

**Review by r0mant — CHANGES_REQUESTED** [review]
I think there's still a bug here. Basically, it looks like `getTLSConfigInsecure` now tries to append client cert in all scenarios, not just when connecting to self-hosted databases.

**Review by r0mant — APPROVED** [review]

**Comment by r0mant** [comment]
Let's backport to v8 and v9.

**Pull request merged**